app/controllers/api/v1/configs_controller.rb

```diff
@@ -0,0 +1,103 @@
+class Api::V1::ConfigsController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :set_config, only: %i[update destroy]
+
+  rescue_from StandardError, with: :handle_standard_error
+  rescue_from ActiveRecord::RecordInvalid, with: :record_invalid
+  rescue_from ActiveRecord::RecordNotUnique, with: :record_not_unique
+  rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+  rescue_from ActionDispatch::Http::Parameters::ParseError, with: :parse_error
+
+  # GET api/v1/configs/name
+  def show
+    config = Config.friendly.find(params[:id])
+
+    render json: config.body, status: :ok
+  end
+
+  # POST api/v1/configs
+  def create
+    config = current_user.configs.create!(get_config)
+
+    render json: {
+             message: "Config #{config.name} was successfully created.",
+             show_instruction: "If you want to see the saved body, send a GET request to /api/v1/configs/#{config.name}"
+           },
+           status: :created
+  end
+
+  # PATCH/PUT api/v1/configs/name
+  def update
+    if params.has_key?(:config) && params[:config].present?
+      @config.update!(get_config)
+    elsif params.has_key?(:"entity types") || params.has_key?(:"relation types")
+      @config.update!(body: get_body)
+    end
+
+    render json: { message: "Config #{@config.name} was successfully updated." }, status: :ok
+  end
+
+  # DELETE api/v1/configs/name
+  def destroy
+    @config.destroy
```

Comment:
```suggestion
    current_config.destroy
```

とすると良さそうです。

1. before_actionを使わなくてもメソッドを呼び出せば同じ事ができる
2. 意味的にset_configよりcurrent_configの方が合いそう

あとconfigが見つからなかった時はどんな動きになりますか？
404を返すと良さそうだと思っています。

Reply:
destroyアクションはconfigの削除とconfig.nameをレスポンスとして返しています。
updateアクションも同様です。

current_configをメソッドとした時、2度呼び出すことになります。
削除後はメソッド呼び出しからconfig.nameを取得できないので変数で保持しておく必要がありそうでした。

したがって、before_actionを使う形で変数名を変える対応をしてみましたが、どうでしょうか？
https://github.com/pubannotation/textae-configs/pull/13/commits/ed7e1b319610c75be8031f3d4a8bc078b20371c9

> あとconfigが見つからなかった時はどんな動きになりますか？
RecordNotFoundの例外処理によってエラーメッセージを404とともに返すようにしています。
```
curl -X DELETE http://localhost:3000/api/v1/configs/abc
{"error":"Could not find the config abc"}

```

```
Started DELETE "/api/v1/configs/abc" for ::1 at 2024-12-04 15:54:38 +0900
Processing by Api::V1::ConfigsController#destroy as */*
  Parameters: {"id"=>"abc"}
  User Load (0.0ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT ?  [["LIMIT", 1]]
  Config Load (0.0ms)  SELECT "configs".* FROM "configs" WHERE "configs"."user_id" = ? AND "configs"."name" = ? LIMIT ?  [["user_id", 1], ["name", "abc"], ["LIMIT", 1]]
Completed 404 Not Found in 4ms (Views: 0.1ms | ActiveRecord: 0.4ms | Allocations: 8224)
```

Reply:
nameだけなら`params[:id]`でとれるので、configインスタンスを参照しなくても良さそうです。

Reply:
レビューありがとうございます。
修正しました。
- configを取得するインスタンス変数をprivate methodに置き換え
  - https://github.com/pubannotation/textae-configs/pull/13/commits/f23d805ad933509befd8a305016d69e8c6caaeca
- id parameterの変更
  - https://github.com/pubannotation/textae-configs/pull/13/commits/db88d0172820c7fe4e3ee7eaa98a42a64c5bcf39